### DISCUSSION TITLE
db:dump

### DISCUSSION THREAD

**Discussion by grafst** [laravel/framework#47906] [Ideas]
I have written a `db:dump` command. Maybe others could also benefit from it....
Where can I make a pullrequest?

```php
<?php

namespace App\Console\Commands;

use Illuminate\Console\Command;
use Illuminate\Support\Facades\Process;


class DumpDatabaseCommand extends Command
{
    protected $signature = 'db:dump {--force : overwrite existing dump}';

    protected $description = 'export the whole database';


    public function handle(): void
    {

        $notice = "Be careful with this file. It could contain sensitive information\n";

        $user = env("DB_USERNAME");
        $database = env("DB_DATABASE");
        $system = env("DB_CONNECTION");
        $host = env("DB_HOST");

        //decide which command to run based on the database system
        switch ($system) {
            case 'mariadb':
            case 'mysql':
                $output_file = $database . ".sql";
                $command = "mysqldump -h $host -u $user -p $database > $output_file";
                break;
            case 'pgsql':
                $output_file = $database . ".tar";
                $command = "pg_dump -h $host -U $user -W -F t $database > $output_file";
                break;
            case 'sqlsrv':
                $output_file = $database . ".bak";
                $command = "Import-Module SqlServer; Backup-SqlDatabase -ServerInstance \"$host\MSSQLSERVER\" -Database \"$database\" -BackupFile \"$output_file\"";
                //todo: implement
                $this->error("sqlsrv not supported yet");
                return;
            case 'sqlite':
                $output_file = "$database.sqlite";
                $input_path = database_path($database . ".sqlite");
                $command = "cp $input_path $output_file";
                break;
            default:
                $this->error("Unknown database system: $system");
                return;
        }

        //prevent dump file from accidentally being overwritten
        if (file_exists($output_file)) {
            if (!$this->option('force')) {
                $this->warn("This command will overwrite $output_file");
                if (!$this->confirm('Do you wish to continue?')) {
                    return;
                }
            }
        }

        //if verbose flag is set: output the command before running it
        if ($this->option('verbose')) {
            $this->info("$command");
        }

        //actually dump the database
        $result = Process::run($command);
        $output = $result->output();
        if ($result->successful()) {
            $this->info("Database dumped to $output_file");
            $this->warn($notice);
        } else {
            $errorOutput = $result->errorOutput();
            $this->error("Dumping the database failed.\n $output \n $errorOutput");
        }
    }
}
```

**Comment by henzeb** [comment]
https://github.com/laravel/framework/pulls

**Reply by grafst** [reply]
Thanks. In what directory do i put the file though?

**Reply by TheBlckbird** [reply]
`src/Illuminate/Console`

**Reply by TheBlckbird** [reply]
And don't forget to change the namespace.

**Comment by devajmeireles** [comment]
> I have written a `db:dump` command. Maybe others could also benefit from it.... Where can I make a pullrequest?
> 
> ```
> <?php
> 
> namespace App\Console\Commands;
> 
> use Illuminate\Console\Command;
> use Illuminate\Support\Facades\Process;
> 
> 
> class DumpDatabaseCommand extends Command
> {
>     protected $signature = 'db:dump {--force : overwrite existing dump}';
> 
>     protected $description = 'export the whole database';
> 
> 
>     public function handle(): void
>     {
> 
>         $notice = "Be careful with this file. It could contain sensitive information\n";
> 
>         $user = env("DB_USERNAME");
>         $database = env("DB_DATABASE");
>         $system = env("DB_CONNECTION");
>         $host = env("DB_HOST");
> 
>         //decide which command to run based on the database system
>         switch ($system) {
>             case 'mariadb':
>             case 'mysql':
>                 $output_file = $database . ".sql";
>                 $command = "mysqldump -h $host -u $user -p $database > $output_file";
>                 break;
>             case 'pgsql':
>                 $output_file = $database . ".tar";
>                 $command = "pg_dump -h $host -U $user -W -F t $database > $output_file";
>                 break;
>             case 'sqlsrv':
>                 $output_file = $database . ".bak";
>                 $command = "Import-Module SqlServer; Backup-SqlDatabase -ServerInstance \"$host\MSSQLSERVER\" -Database \"$database\" -BackupFile \"$output_file\"";
>                 //todo: implement
>                 $this->error("sqlsrv not supported yet");
>                 return;
>             case 'sqlite':
>                 $output_file = "$database.sqlite";
>                 $input_path = database_path($database . ".sqlite");
>                 $command = "cp $input_path $output_file";
>                 break;
>             default:
>                 $this->error("Unknown database system: $system");
>                 return;
>         }
> 
>         //prevent dump file from accidentally being overwritten
>         if (file_exists($output_file)) {
>             if (!$this->option('force')) {
>                 $this->warn("This command will overwrite $output_file");
>                 if (!$this->confirm('Do you wish to continue?')) {
>                     return;
>                 }
>             }
>         }
> 
>         //if verbose flag is set: output the command before running it
>         if ($this->option('verbose')) {
>             $this->info("$command");
>         }
> 
>         //actually dump the database
>         $result = Process::run($command);
>         $output = $result->output();
>         if ($result->successful()) {
>             $this->info("Database dumped to $output_file");
>             $this->warn($notice);
>         } else {
>             $errorOutput = $result->errorOutput();
>             $this->error("Dumping the database failed.\n $output \n $errorOutput");
>         }
>     }
> }
> ```

Thinking at the tool level, this is useless for Laravel, but useful as a third-party package tool. Similar to this you can use: https://spatie.be/docs/laravel-backup/v8/introduction
